pyomo/util/tests/test_latex_printer.py

```diff
@@ -0,0 +1,791 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2023
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+import io
+import pyomo.common.unittest as unittest
+from pyomo.util.latex_printer import latex_printer
+import pyomo.environ as pyo
+from textwrap import dedent
+from pyomo.common.tempfiles import TempfileManager
+from pyomo.common.collections.component_map import ComponentMap
+
+from pyomo.environ import (
+    Reals,
+    PositiveReals,
+    NonPositiveReals,
+    NegativeReals,
+    NonNegativeReals,
+    Integers,
+    PositiveIntegers,
+    NonPositiveIntegers,
+    NegativeIntegers,
+    NonNegativeIntegers,
+    Boolean,
+    Binary,
+    Any,
+    # AnyWithNone,
+    EmptySet,
+    UnitInterval,
+    PercentFraction,
+    # RealInterval,
+    # IntegerInterval,
+)
+
+
+def generate_model():
+    import pyomo.environ as pyo
+    from pyomo.core.expr import Expr_if
+    from pyomo.core.base import ExternalFunction
+
+    m = pyo.ConcreteModel(name='basicFormulation')
+    m.x = pyo.Var()
+    m.y = pyo.Var()
+    m.z = pyo.Var()
+    m.objective_1 = pyo.Objective(expr=m.x + m.y + m.z)
+    m.constraint_1 = pyo.Constraint(
+        expr=m.x**2 + m.y**-2.0 - m.x * m.y * m.z + 1 == 2.0
+    )
+    m.constraint_2 = pyo.Constraint(expr=abs(m.x / m.z**-2) * (m.x + m.y) <= 2.0)
+    m.constraint_3 = pyo.Constraint(expr=pyo.sqrt(m.x / m.z**-2) <= 2.0)
+    m.constraint_4 = pyo.Constraint(expr=(1, m.x, 2))
+    m.constraint_5 = pyo.Constraint(expr=Expr_if(m.x <= 1.0, m.z, m.y) <= 1.0)
+
+    def blackbox(a, b):
+        return sin(a - b)
+
+    m.bb = ExternalFunction(blackbox)
+    m.constraint_6 = pyo.Constraint(expr=m.x + m.bb(m.x, m.y) == 2)
+
+    m.I = pyo.Set(initialize=[1, 2, 3, 4, 5])
+    m.J = pyo.Set(initialize=[1, 2, 3])
+    m.K = pyo.Set(initialize=[1, 3, 5])
```

Comment:
Have you tested the printer on Sets containing non-numeric entries like strings?

Reply:
No, but in theory it shouldn't matter what's in the set unless you use the `explicit_set_summation` flag.  There is a known issue with how things like pro.Set()*pyo.Set() end up rendering, but I'm living with that for now